internal/payments/service.go

```diff
@@ -0,0 +1,218 @@
+type CreateProjectInvoiceParams struct {
+	ProjectName     string
+	CustomerID      string
+	PaymentMethodID string
+
+	Storage     float64
+	Egress      float64
+	ObjectCount float64
+
+	StartDate time.Time
+	EndDate   time.Time
+}
+
+// NewService creates new instance of StripeService initialized with API key
+func NewService(log *zap.Logger, apiKey string) *StripeService {
+	stripe.DefaultLeveledLogger = wrapLogger(log)
+
+	sc := new(client.API)
+	sc.Init(apiKey, nil)
+
+	return &StripeService{
+		log:    log,
+		client: sc,
+	}
+}
+
+// CreateCustomer creates new customer from CustomerParams struct
+// sets default payment to one of the predefined testing VISA credit cards
+func (s *StripeService) CreateCustomer(ctx context.Context, params CreateCustomerParams) (*stripe.Customer, error) {
+	cparams := &stripe.CustomerParams{
+		Email:       stripe.String(params.Email),
+		Name:        stripe.String(params.Name),
+		Description: stripe.String(params.Description),
+	}
+
+	// Set default source (payment instrument)
+	//if params.SourceToken != "" {
+	//	err := cparams.SetSource(params.SourceToken)
+	//	if err != nil {
+	//		return nil, StripeErr.Wrap(err)
+	//	}
+	//}
+
+	// TODO: delete after migrating from test environment
+	err := cparams.SetSource("tok_visa")
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Customers.New(cparams)
+}
+
+// GetCustomer retrieves customer object from stripe network
+func (s *StripeService) GetCustomer(ctx context.Context, id string) (*stripe.Customer, error) {
+	return s.client.Customers.Get(id, nil)
+}
+
+// GetCustomerDefaultPaymentMethod retrieves customer default payment method from stripe network
+func (s *StripeService) GetCustomerDefaultPaymentMethod(ctx context.Context, customerID string) (*stripe.PaymentMethod, error) {
+	cus, err := s.client.Customers.Get(customerID, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if cus.DefaultSource == nil {
+		return nil, StripeErr.New("no default payment method attached to customer")
+	}
+
+	return s.client.PaymentMethods.Get(cus.DefaultSource.ID, nil)
+}
+
+// GetCustomerPaymentsMethods retrieves all payments method attached to particular customer
+func (s *StripeService) GetCustomerPaymentsMethods(ctx context.Context, customerID string) ([]*stripe.PaymentMethod, error) {
+	var err error
+
+	pmparams := &stripe.PaymentMethodListParams{}
+	pmparams.Filters.AddFilter("customer", "", customerID)
+	pmparams.Filters.AddFilter("type", "", "card")
+
+	iterator := s.client.PaymentMethods.List(pmparams)
+	if err = iterator.Err(); err != nil {
+		return nil, err
+	}
+
+	var paymentMethods []*stripe.PaymentMethod
+	for iterator.Next() {
+		pm := iterator.PaymentMethod()
+		paymentMethods = append(paymentMethods, pm)
+	}
+
+	return paymentMethods, nil
+}
+
+// GetPaymentMethod retrieve payment method object from stripe network
+func (s *StripeService) GetPaymentMethod(ctx context.Context, id string) (*stripe.PaymentMethod, error) {
```

Comment:
Is the `id` here a paymentMethod id from stripe?

Reply:
yes